rfc/0006-kubewarden-versioning.md

```diff
@@ -0,0 +1,203 @@
+    version: v0.4.8
+  - component: policy-server
+    version: v0.3.0
+  - component: kwctl
+    version: v0.3.1
+...
+```
+
+As more components are relevant they can be added to the list as time
+grows.
+
+This map allows us to generate documentation and a single source of
+truth to refer to when a user has a problem, or when we are debugging
+an issue. It helps to understand if any component has drifted from the
+version we are expecting.
+
+The helm chart will be responsible of storing the "Kubewarden version"
+in a resource, like a ConfigMap. So it's easy to inspect it and
+understand what "Kubewarden version" a cluster is running.
+
+Had other external components such as `kwctl` need to synchronize with
+the cluster, they can inspect this ConfigMap and realize if they are
+compatible with the running version of Kubewarden on the cluster they
+are connecting to.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+- It's not a very common versioning scheme. I would consider reading
+  ["When to use
+  CalVer"](https://calver.org/#when-to-use-calver). Although
+  Kubewarden does not have a "large" scope, it's to some extent
+  "constantly changing", and this constant change would imply
+  harmonizing versions of all components continuously.
+
+# Alternatives
+[alternatives]: #alternatives
+
+## Have a regular Kubewarden toplevel version
+
+Having a regular --even semver based-- version on the toplevel project
+seems like an artificial construct. What is the meaning of patchlevel
+release in this case? What is Kubewarden 1.2.3 vs 1.2.4?
+
+Thus, ignoring patchlevel (e.g. just keeping 1.2 vs 1.3 vs 1.4, or
+even 1 vs 2 vs 3) is an option. But then, if it's just meaningless
+numbers, why not make the toplevel number meaningful, referring to the
+date when the project was released. This brings instant information to
+the user as opposed to version numbers.
+
+## Sync all component versions
+
+Given the nature of multiple components of the Kubewarden projects,
+scattered across different repositories, closely related to each
+other, enforcing a single synced version on all of them seem
+artificial and heavily increases releasing efforts, given a problem or
+feature in on one component forces to tag and release the rest of the
+components.
+
+# Unresolved questions
+[unresolved]: #unresolved-questions
+
+- Should day versions be included in the toplevel map?
+
+    To make things even easier, versions might start with
+    YEAR.MONTH. If there is the need to make a second global release
+    during the same month, then a day is appended: YEAR.MONTH.DAY1. If
+    a third release is needed during that month: YEAR.MONTH.DAY2.
+
+    This makes it easier for maintainability of upgrades and is easier
+    to describe what should the upgrade path be if there is the
+    necessity of forcefully sequence upgrades in a certain order.
+
+- Should complete versions of components (MAJOR.MINOR.PATCH) be
+  referenced by the versions map as the example shows?
```

Comment:
I vote for complete versions of components, i.e., if we release a new patch for controller, we bump the general version of Kubewarden, and bump the day version.